### DISCUSSION TITLE
Help to implement Fly-Pie in hyprland

### DISCUSSION THREAD

**Discussion by agdral** [Schneegans/Fly-Pie#287] [Ideas & Feature Discussions]
Hi, I am writing a program that seeks to implement the same functionality as fly-pie but for hyprland (A dynamic tiling Wayland composer), this composer solves the limitations listed [here](https://github.com/Schneegans/Fly-Pie/issues/22). The problem I've found is that I can't understand how you managed to create the visual part of fly pie in gtk, things like: the positioning of the items, the drag and drop function, where the rendering functions are executed ?

**Comment by Schneegans** [comment]
Hi there! This is a great idea, however I am not sure what kind of help you would expect here. Fly-Pie is open source, so feel free to study the code. If you have concrete questions I can give you some hints regarding where to look for a specific piece of code.

Yet I am not sure how much this will help, as Fly-Pie uses Clutter for drawing the menu and not GTK. So all the rendering and interaction code will be different...

**Comment by agdral** [comment]
thanks, my question was exactly that, I didn't know what clutter was in the code and I passed it by, now I'll see what I can do with that knowledge, if I find myself confused with something in the code I'll let you know.

**Comment by agdral** [comment]
Hello, it's been a while since we spoke. I have  a question. i want to use the application gui of configuration in Fly-Pie but the file.ui depend of functions in code javascript. My question is if you have a version pure of file.ui (that no use functions for work)?

**Reply by Schneegans** [reply]
I have only the files in the repository. And they are really hard to work with! I would definitely recommend to start with a small custom ui file and then slowly add features to it. You can definitely draw inspiration from Fly-Pie's ui files, but copying them as they are will be very difficult, I assume...

**Comment by Schneegans** [comment]
Btw, I recently started working on a [platform-agnostic pie menu](https://ko-fi.com/post/Introducing-Ken-Do-L3L7L0FQ2)! This will most likely also work in hyprland once it's finished.

**Comment by agdral** [comment]
thanks for reply, i am trying to recreate the menu config with my needed, but your new menu is interesting. i going to see that i can do
